### PR TITLE
MINOR: Move RocksDBTimeOrderedKeyValueBufferTest to use Junit5

### PR DESCRIPTION
**Pull request by wcarlson5**
Upgrade RocksDBTimeOrderedKeyValueBufferTest to use Junit5

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java [hunk 1]
```diff
@@ -32,31 +32,28 @@
...
+import org.junit.jupiter.api.Test;
 
 import java.time.Duration;
 import java.util.concurrent.atomic.AtomicInteger;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
```
Comment (divijvaidya) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java:47]: Why did we remove this? StrictStubs help catch unnecessary stubs and keeps the tests clean.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java [hunk 2]
```diff
@@ -32,31 +32,28 @@
...
 public class RocksDBTimeOrderedKeyValueBufferTest {
     public RocksDBTimeOrderedKeyValueBuffer<String, String> buffer;
-    @Mock
```
Comment (divijvaidya) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java:50]: Why did we change this? `@Mock` annotation should work with JUnit5?

**Review by divijvaidya — COMMENTED** [review]
I verified that the test is still running in CI : https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13878/1/testReport/org.apache.kafka.streams.state.internals/RocksDBTimeOrderedKeyValueBufferTest/ so the change functionally looks good. Left some minor comments.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java (hunk 2 above)
Comment (wcarlson5) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java:50]: It wasn't working, but the mock() was

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java (hunk 1 above)
Comment (wcarlson5) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java:47]: It shouldn't be necessary, it already fails unused stubs

Comment (cadonna) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java:47]: I tested this and I can confirm what Walker stated.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java (hunk 2 above)
Comment (cadonna) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java:50]: To get `@Mock` to work with JUnit 5, you need to apply `@ExtendWith(MockitoExtension.class)` to the test class.
That is:
```java
@ExtendWith(MockitoExtension.class)
public class RocksDBTimeOrderedKeyValueBufferTest {
...
```

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java [hunk 3]
```diff
@@ -32,31 +32,28 @@
...
     public StreamsMetricsImpl streamsMetrics;
-    @Mock
     public Sensor sensor;
     public long offset;
 
-    @Before
+    @BeforeEach
     public void setUp() {
+        serdeGetter = mock(SerdeGetter.class);
+        sensor = mock(Sensor.class);
```
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBufferTest.java:56]: I could not find any usage of this mock.

**Review by cadonna — COMMENTED** [review]
Thanks @wcarlson5 for the PR!

Here my feedback!

**Pull request closed without merging**